### PR TITLE
Convert more API Documentation to Open API

### PR DESCRIPTION
**Pull request by lyzadanger**
Good progress! With this PR, I've done the following to our new (as yet invisible) Open API docs:

* Added annotation endpoint documentation
* Added group-membership documentation
* Added annotation moderation endpoint documentation
* Added the service root endpoint documentation
* Expanded and updated the error documentation
* Started to flesh out the Introductory content (WIP)

Much of this is ported verbatim from our existing docs with some cleanup (and compatibility changes).

The annotation response documentation is inaccurate, but it's the same as our current docs and I'm not going to jump into that just now...there are a few other shortcomings that we can address as we go...

I believe the only endpoint to add to these new docs is the search endpoint. Getting there! 

Part of https://github.com/hypothesis/product-backlog/issues/946

### CODE REVIEW

File: docs/_extra/api-reference/hypothesis-v1.yaml [hunk 1]
```diff
@@ -125,6 +197,10 @@ components:
   # Resusable resource schemas
   # ------------------------------
   schemas:
+    Annotation:
+      $ref: './schemas/annotation-schema.json'
+    AnnotationCreate:
+      $ref: './schemas/annotation-schema.json'
```
Comment (robertknight) [inline docs/_extra/api-reference/hypothesis-v1.yaml]: These two types definitely shouldn't be the same. Is that what you meant about inaccuracies in the annotation response? For example, the docs for the annotation response (eg. `GET /annotations/{id}`) don't mention the ID or timestamp fields (!)

File: docs/_extra/api-reference/hypothesis-v1.yaml [hunk 2]
```diff
@@ -67,24 +116,47 @@ components:
...
+    UserID:
+      name: user
+      in: path
+      required: true
+      description: >
+        Unique identifier for a user, with specified authority.
+        The userID should be of the format `acct:<username>@<authority>`
+      schema:
+        type: string
+        pattern: "acct:(?i)^[A-Z0-9._]+${3,30}@.*$"
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply docs/_extra/api-reference/hypothesis-v1.yaml]: Me being lazy here: I c-n-p'ed that from the `user` model. I'll take another look and be more accurate.

File: docs/_extra/api-reference/hypothesis-v1.yaml [hunk 3]
```diff
@@ -67,24 +116,47 @@ components:
...
+    BadRequest:
+      description: Bad Request
+      content:
+        application/*json:
+          schema:
+            $ref: './schemas/errors.yaml#/Error'
     Conflict:
-      description: >
-        An error indicating a resource conflict, typically because a
-        duplicate resource already exists.
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply docs/_extra/api-reference/hypothesis-v1.yaml]: Concision. But I can put it back!

File: docs/_extra/api-reference/hypothesis-v1.yaml [hunk 4]
```diff
@@ -146,6 +222,228 @@ components:
...
+        - ApiKey: []
+      parameters:
+        - $ref: '#/components/parameters/AnnotationID'
+      responses:
+        '204':
+          $ref: '#/components/responses/NoContent'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '415':
+          $ref: '#/components/responses/UnsupportedMediaType'
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply docs/_extra/api-reference/hypothesis-v1.yaml]: Yeah, I was debating this back and forth myself. The benefit of doing it like this is that the schema is "real" and machine-readable. The obvious benefit for doing it in a single, introductory location is that it's less repetition and gives us the ability to flesh out the documentation beyond basic schema fields. But we lose the schema itself. I wonder if _both_ might be the best approach (real schema on the endpoints, explanatory intro docs). Thoughts?

File: docs/_extra/api-reference/hypothesis-v1.yaml [hunk 5]
```diff
@@ -316,10 +629,74 @@ paths:
...
+    delete:
+      tags:
+        - groups
+      summary: Remove member from group
+      description: >
+        Remove a user from a group. At present, this endpoint only allows
+        the removal as one's self (authenticated with API Key) from the
+        indicated group.
+      security:
+        - AuthClient: []
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply docs/_extra/api-reference/hypothesis-v1.yaml]: Good catch. That's straight-up wrong. I blame c-n-p fatigue and EOD loopiness.

File: docs/_extra/api-reference/hypothesis-v1.yaml (hunk 1 above)
Comment (lyzadanger) [reply docs/_extra/api-reference/hypothesis-v1.yaml]: This is indeed what I mean about the `annotation` schema being completely wrong in our current docs. These new docs share the same underlying resource schema (for now, at least). Yeah, it needs fixing, but it's orthogonal.

File: docs/_extra/api-reference/hypothesis-v1.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (lyzadanger) [reply docs/_extra/api-reference/hypothesis-v1.yaml]: Here's an example of how Clearbit handles their error documentation. They have both a prose-documentation section (https://clearbit.com/docs?python#errors) and then they also have per-endpoint error information. I think I might go this general direction.

**Comment by lyzadanger** [comment]
@robertknight I fleshed out the beginnings of an errors (and versions) section in the general documentation, and I fixed the glitches you noted.

File: docs/_extra/api-reference/hypothesis-v1.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (robertknight) [reply docs/_extra/api-reference/hypothesis-v1.yaml]: I think it is useful to include error codes _that are specific to the endpoint_ in the docs for a given endpoint, but for errors which can happen for any request (eg. invalid API keys, unknown API version) I think following Clearbit's example, to just document them in the general API docs, would make sense.

**Pull request merged**